### ISSUE TITLE
Sitemap xml generating problem - no main language pages included

### ISSUE THREAD

**Issue by qbadev** [exploreimpact/contao-i18nl10n#82]
Contao 3.3.4, i18nl10n 1.5.3
As in subject, only German and Polish pages are included in Sitemap.xml. Main lanugage (english) pages cannot be found inside file mentioned above.

**Comment by patric-eberle** [comment]
Was not able to reproduce this. Still, there was an issue, which excluded some pages from indexing. But the main language was always indexed. The module also does not change anything of the indexation of the main language and only adds additional entries. Are you sure this is caused by the i18nl10n module?

**Comment by qbadev** [comment]
Before installing i18nl10n  there were only english pages. And sitemap.xml was OK. After installing and adding polish and german language, sitemap includes only german pages and some of polish ones. There is a Hook in getSearchablePages and generateFrontendUrl. 
I was digging a bit lately, and my first conclusion was that there must be a problem in generateFrontendUrl hook, because before this function records (pages) where fine (english), and after this it became only polish and german.
I will continue my research and give you "full report"

**Comment by patric-eberle** [comment]
I just discovered, that it seems to have an impact which language is defined for the backend user...

**Comment by qbadev** [comment]
I had that feeling as well, but I changed backend interface lang and sitemap was still incorrect with no changes...

**Comment by internetinnovations** [comment]
Most probably we will integrate our new sitemap extension into the i18nl10n. That might also solve this main language page issue.

**Comment by paulhaem** [comment]
Any news regarding the sitemap.xml issue? I am using English as main language and German as second language. The sitemap.xml does only generate German pages. For big pages it's quite important to provide a sitemap.xml for Google.

**Comment by paulhaem** [comment]
In addition: currently the sitemap.xml is not generated by definition. The URLs are inside of an `loc`-Tag. The additional `xhtml:link`-Tag for the self-reference is missing.


```
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd">
<url>
	<loc>../home.html</loc>
</url>
<url>
	<loc>../about.html</loc>
</url>
```

Reference: https://support.google.com/webmasters/answer/2620865?hl=en

**Comment by internetinnovations** [comment]
As mentioned above, we've developed a new sitemap extension for i18nl10n.
Last week, we've started refactoring i18nl10n for Contao 4.4
Current Contao 3.5 versions will not be supported anymore with i18nl10n version 2.0.
Our sitemap extension will be included in i18nl10n version 2.0.
The beta testing phase might be possible in September 2017.
@paulhaem

**Comment by paulhaem** [comment]
You mean to say, that all bugfixes are Contao 4.4 exclusive?

**Comment by internetinnovations** [comment]
Right now this is correct. During 2016 we've internally implemented many optimizations for this extension. We might be able to push them in addition for 3.5.28 after the release of version 2.0 for Contao 4.4
@paulhaem

**Comment by paulhaem** [comment]
It would be really awesome to have those fixes for Contao 3.5.28. There are a lot of production websites using Contao 3.5.28 in combination with i18nl10n waiting for those fixes since a long time and I guess many would appreciate this.

**Comment by Aybee** [comment]
Same problem here. Does anyone know a workaround for Contao 3.5?

I have also noticed that the main language is sometimes in there and sometimes not.
